### ISSUE TITLE
Player is not saving after exit to lobby

### ISSUE THREAD

**Issue by IvanMMM** [micovery/ArmA3_Wasteland.Altis#21]
Hey.
Player can just exit to lobby and reenter and he will get his old data loaded. That's terrible

**Comment by codexGW** [comment]
I think that's vanilla behavior as well.

**Comment by IvanMMM** [comment]
Not really. If player calls main menu (Esc) his player is force saved. Vanilla issue is alt+f4 and force disconnect.

**Comment by micovery** [comment]
Going back to lobby is equivalent to disconnecting

**Comment by IvanMMM** [comment]
Well, anyway it's not working as needed. I can create video if you want.

**Comment by micovery** [comment]
Scenario #1:
You Ctrl + Alt + Del, and and kill your Arma 3 process, and then reconnect to the server, and now you are back in lobby.

Scenario #2:
Suppose that you are kicked by the admin, and then you reconnect to the server, and you are back into lobby.

Scenario #3:
You press ESC, and click on abort, and you are back into lobby.

These are all just different ways to  "quit" the game, but the one thing they all have in common, is that there is always a "disconnect" event.  The "vanilla" mission only handles Scenario #3. My fork handles any possible way you think think of that results in a  "disconnect" event.

There are some special cases where stats should not be saved. These are: disconnecting while unconscious,  and disconnecting from the "Respawn dialog".

Can you elaborate more on what the issue is?

**Comment by IvanMMM** [comment]
I'm using exit with esc.
Move somewhere, exit to lobby, enter game, move somewhere, exit to lobby, enter game - your newest location is not saved.

**Comment by micovery** [comment]
That would be a bug, you location should be restored from your "last" disconnect event. 

Take a look at:

https://github.com/micovery/ArmA3_Wasteland.Altis/blob/Development_main/persistence/players/sFunctions.sqf#L165

I imagine, that flag is not being set back to true when you go back to lobby, and re-join multiple times.

**Comment by micovery** [comment]
I am not able to reproduce the issue with the latest code.

I do this:
1. Join server, and Spawn
2. Press Esc, Click Abort, Confirm Yes, Click Disconnect, Arrive at lobby
3. From lobby, Click OK, and Spawn
4. Move to a new location,  And Go to Step #2 

I Keep doing that, and each time, I puts me back to the correct location (where I was before pressing Esc).

Are you doing something differently?

**Comment by micovery** [comment]
I have pushed code through 30719e73e2085c95436a56f75fe8ceacfdf50c0a, to add extra logging when a player disconnects. You should see a warning like

```
WARNING: No stats saved for player1(3454423423423) on disconnect"
```

If no stats are saved.

**Comment by IvanMMM** [comment]
Hm. Well, it mb just my error.  But I found direction is always reseted :)

**Comment by micovery** [comment]
Yep, direction is not restoring (or maybe not saving correctly), I'll investigate. Thanks for the report on that.

**Comment by micovery** [comment]
Yah, it's another duh moment :-)  when I made this change

40542a2c2d4ec8b6a03361653aadd5f2ed5181ff

I forgot to actually set the position, and the direction. I had fixed the position in the next commit, but did not fix the direction. Fixing this now.

**Comment by micovery** [comment]
The missing direction is now fixed by 08cac15e068774c3d362ee029744daa5a86e29be

I am closing this issue for now as not reproducible. Please re-open it if you are able to reproduce this with the latest code.
